c2corg_ui/static/js/suggestiondirective.js

```diff
@@ -0,0 +1,50 @@
+goog.provide('app.suggestionDirective');
+
+/**
+ * @param {angular.$compile} $compile Angular compile service.
+ * @param {angular.$sce} $sce Angular Strict Contextual Escaping service.
+ * @return {angular.Directive} Directive Definition Object.
+ * @ngInject
+ */
+app.suggestionDirective = function($compile, $sce) {
+  var header = '<span class="suggestion-text">' +
+                            '<span class="suggestion-title" ng-bind-html="highlight(label, query)"></span><br>' +
+                            '<span class="suggestion-info">' +
+                            '  <span ng-if="elevation != \'null\' ">{{::elevation}}m</span>' +
+                            '</span>';
+  var footer =   '</span>';
+
+  var template = function(doctype, activities) {
+    if (doctype === 'waypoints') {
+      return header + '</span><span class="icon-container"><span class="icon-{{::type}}"></span>' + footer;
+    } else if (doctype === 'routes' || doctype === 'outings') {
+      var activitiesHtml = '';
+      var acts = activities.split(',');
+      for (var i = 0; i < acts.length; i++) {
+        activitiesHtml += '<span class="icon-' + acts[i] + '"></span>'
+      }
+      return header + '</span><span class="icon-container">' + activitiesHtml + footer;
+    }
+  }
+
+  return {
+    restrict: 'E',
+    scope: {'type': '@', 'label': '@', 'dataset': '@', 'elevation': '@', 'activities': '@', 'query': '@'},
```

Comment:
Indeed that's what I told myself after there were a bunch of them. I will try to get rid of this when I come back to this pr.